src/Options/Applicative/Help/Types.hs

```diff
@@ -42,6 +53,44 @@ helpText (ParserHelp e s h u d b g f) =
 -- | Convert a help text to 'String'.
 renderHelp :: Int -> ParserHelp -> String
 renderHelp cols
-  = (`displayS` "")
-  . renderPretty 1.0 cols
+  = LT.unpack
+  . B.toLazyText
+  . renderAnsi
+  . layoutPretty (LayoutOptions (AvailablePerLine cols 1.0))
   . helpText
+
+renderAnsi :: SimpleDocStream Ann -> B.Builder
+renderAnsi
+  = renderCtxDecorated defaultStyle B.fromText renderPush renderPop
+  . alterAnnotationsS alter
+  where
+    alter :: Ann -> Maybe SetStyle
+    alter (AnnStyle setStyle) = Just setStyle
+    renderPush :: SetStyle -> SetStyle -> B.Builder
+    renderPush _ setStyle = B.fromString (styleToRawText setStyle)
+    renderPop :: SetStyle -> SetStyle -> B.Builder
+    renderPop setStyle _ = B.fromString (styleToRawText setStyle)
+
+renderCtxDecorated
+    :: Monoid ann
+    => ann
+    -> (Text -> B.Builder) -- ^ Render plain 'Text'
+    -> (ann -> ann -> B.Builder)  -- ^ How to render an annotation
+    -> (ann -> ann -> B.Builder)  -- ^ How to render the removed annotation
+    -> SimpleDocStream ann
+    -> B.Builder
+renderCtxDecorated topAnn toText push pop = go [topAnn]
+  where
+    go _                      SFail               = panicUncaughtFail
+    go []                     SEmpty              = mempty
+    go (_:_:_)                SEmpty              = panicInputNotFullyConsumed
+    go (_:_)                  SEmpty              = mempty
+    go stack                  (SChar c rest)      = toText (T.singleton c) <> go stack rest
+    go stack                  (SText _l t rest)   = toText t <> go stack rest
+    go stack                  (SLine i rest)      = toText (T.singleton '\n') <> toText (textSpaces i) <> go stack rest
+    go stack@(ctxAnn:_)       (SAnnPush ann rest) = push ctxAnn ann <> go ((ctxAnn <> ann) : stack) rest
+    go (ann:stack@(ctxAnn:_)) (SAnnPop rest)      = pop ctxAnn ann <> go stack rest
+    go []                     SAnnPop{}           = panicUnpairedPop
+    go _                      (SAnnPush _ _)      = error "Unpaired push"
+    go _                      (SAnnPop _)         = error "Unpaired pop"
```

Comment:
Sorry, I can't accept something like this function.

We really can't be doing full pattern matches on the internals of dependant libraries in something which sits close to the bottom of most applications dependency trees.

Reply:
I would like to suggest this isn't dependent on internals of libraries because the `prettyprinter` library has tutorials showing describing code very much like this which means to me this interface is public.

For example: https://github.com/quchen/prettyprinter/blob/9d7a40f2be5ce9a8e9c993b683e01d76eb06a079/prettyprinter/src/Prettyprinter/Render/Tutorials/StackMachineTutorial.hs#L99

Reply:
Perhaps @quchen can comment?